packages/bp/src/core/security/router-security.ts

```diff
@@ -178,12 +178,14 @@ const checkPermissions = (workspaceService: WorkspaceService) => (
     })
   }
 
+  const isBotIdValid = req.params.botId && req.params.botId !== ALL_BOTS
+
   if (!req.tokenUser) {
     audit(debugFailure, { email: 'n/a', reason: 'unauthenticated' })
     return new ForbiddenError('Unauthorized')
   }
 
-  if (!req.workspace && req.params.botId) {
+  if (!req.workspace && isBotIdValid) {
```

Comment:
I think we should roll back that change so that if we call this in a middleware where the botId is `___`, it would set the workspace to `default`.